### PR TITLE
Payment gateway orders list

### LINKED ISSUE #40: Integrate payment with product
Payment gateway should be integrated with product on clicking buy now button.

### PR DESCRIPTION
**Pull request by vignesh-s**
-Fixes #40 , #42 
PayU payment gateway implemented
  -user can buy products by using buy button in product details
  -shipping address have to provide only if requiedShipping true
  -can choose any one of payment modes - Credit card, Debit card, NetBanking
  -then paymentGateway webView will come
  -after successfull payment respective message will display & can take
  exam (if package includes) or can continue purchase
  -if payment failed at any stage can retry payment
Orders List & Product List
 -orders list will be shown by clicking orders in main menu
 -earlier product is displayed in cards, now it is changed to ListView
 -free orders can be done
screenshot_2015-09-26-20-00-41
screenshot_2015-09-26-20-00-53
screenshot_2015-09-26-20-00-59
screenshot_2015-09-26-20-01-13
screenshot_2015-09-26-19-59-51
screenshot_2015-09-26-20-00-00
screenshot_2015-09-26-20-00-15

### CODE REVIEW

**Comment by vignesh-s** [comment]
i used product slag as item name in order because order item contains only product url.
getting the product details using url & get the title from it is not a good method, because for each list item there will a server req & response.
so instead giving product url in order item , we can give product itself

File: app/build.gradle [hunk 1]
```diff
@@ -18,46 +18,39 @@ repositories {
...
     // Android Support Libs
     compile 'com.android.support:support-v4:22.0.0'
     compile 'com.android.support:appcompat-v7:22.0.0'
-
     //SlidingUp Panel
     compile 'com.sothree.slidinguppanel:library:3.0.0'
-
     compile 'com.joanzapata.android:android-iconify:1.0.9'
     compile 'com.astuetz:pagerslidingtabstrip:1.0.1'
-
```
Comment (bharathwaaj) [inline app/build.gradle]: Why empty lines removed? They are there to not to make this file look cluttered. Please put it back.

File: app/src/main/java/in/testpress/testpress/core/ProductService.java [hunk 2]
```diff
@@ -19,6 +23,15 @@
     @GET("/{product_url}")
     ProductDetails getProductDetails(@EncodedPath("product_url") String productUrlFrag, @Header("Authorization") String authorization);
 
+    @POST(Constants.Http.URL_ORDERS_FRAG)
+    Order order(@Body HashMap<String, Object> arguments, @Header("Authorization") String authorization);
+
+    @PUT(Constants.Http.URL_ORDERS_FRAG+"{order_id}"+"/confirm/")
```
Comment (bharathwaaj) [inline app/src/main/java/in/testpress/testpress/core/ProductService.java]: This is not the place to calculate the url. It is better to pass it like product_url.

File: app/src/main/java/in/testpress/testpress/ui/OrderConfirmActivity.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+                confirmOrder();
+            } else {
+                internetConnectivityChecker.showAlert();
+            }
+        }
+    }
+
+    private boolean validate() {
+        //Phone number verification
+        Pattern phoneNumberPattern = Pattern.compile("[789]\\d{9}");
```
Comment (bharathwaaj) [inline app/src/main/java/in/testpress/testpress/ui/OrderConfirmActivity.java]: Remove 789

File: app/src/main/java/in/testpress/testpress/ui/OrderConfirmActivity.java [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+            @Override
+            public Order call() throws Exception {
+                return serviceProvider.getService(OrderConfirmActivity.this).orderConfirm(order.getId(),address.getText().toString(), zip.getText().toString(), phone.getText().toString(), landmark.getText().toString(), Constants.Http.URL_USERS + "testpress/", order.getOrderItems());
+            }
+
+            @Override
+            protected void onSuccess(Order order) {
+                Intent intent = new Intent(OrderConfirmActivity.this, PaymentModeActivity.class);
+                paymentParams = new PaymentParams();
+                paymentParams.setKey("72G8r5");
```
Comment (bharathwaaj) [inline app/src/main/java/in/testpress/testpress/ui/OrderConfirmActivity.java]: This should be from order output.

File: app/src/main/res/layout/payment_success_layout.xml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_centerHorizontal="true"
+                android:textColor="#009933"
+                android:layout_below="@id/orderId"/>
+
+            <TextView
+                android:id="@+id/bookMessage"
+                android:textSize="15sp"
+                android:text="Books will be Delivered to you shortly"
```
Comment (adaikalaraj) [inline app/src/main/res/layout/payment_success_layout.xml]: Remove hard coded text and use strings in android

**Comment by bharathwaaj** [comment]
Please give separate pull request for #40 and #42

**Pull request closed without merging**